### PR TITLE
[nlohmann_json] New build recipe

### PR DESCRIPTION
**Pull request by abussy**
New build recipe for https://github.com/nlohmann/json, a JSON library for modern C++.

### CODE REVIEW

**Comment by imciner2** [comment]
Since this is a header-only library, is there a reason to not just bundle the versions with any consumers of it? It seems somewhat overkill to build a JLL on every platform to just provide the header for this.

**Comment by giordano** [comment]
Also, for header-only library targeting `AnyPlatform` is usually a better idea: https://docs.binarybuilder.org/stable/building/#Platform-independent-packages

File: N/nlohmann_json/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir
+
+mkdir build
+cd build
+
+CMAKE_ARGS="-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+            -DCMAKE_FIND_ROOT_PATH='${prefix}/lib/mpich;${prefix}' \
```
Comment (giordano) [inline N/nlohmann_json/build_tarballs.jl:19]: Why `${prefix}/lib/mpich`?

Comment (abussy) [reply N/nlohmann_json/build_tarballs.jl:19]: An artifact left over from the recipe I copied from, my bad

File: N/nlohmann_json/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "nlohmann_json"
```
Comment (giordano) [inline N/nlohmann_json/build_tarballs.jl:5]: Not a fan of this name, but I guess `json` would be incredibly generic.  Is this packaged in any Linux distribution?  If so, how they call it there?

Comment (eschnett) [reply N/nlohmann_json/build_tarballs.jl:5]: That's the package's regular name. See e.g. https://pypi.org/project/nlohmann-json/ or https://packages.ubuntu.com/search?keywords=nlohmann-json3-dev .

Comment (giordano) [reply N/nlohmann_json/build_tarballs.jl:5]: That's it then.

**Pull request merged**